telegram-bot-bumblebee-core/src/main/java/com/github/bumblebee/command/youtube/service/YoutubeSubscriptionService.java

```diff
@@ -0,0 +1,80 @@
+package com.github.bumblebee.command.youtube.service;
+
+import com.github.bumblebee.bot.BumblebeeConfig;
+import com.github.bumblebee.command.youtube.api.YoutubeSubscriptionApi;
+import com.github.bumblebee.command.youtube.dao.YoutubeSubscriptionRepository;
+import com.github.bumblebee.command.youtube.entity.Chat;
+import com.github.bumblebee.command.youtube.entity.Subscription;
+import com.google.api.client.util.Lists;
+import feign.Feign;
+import feign.Response;
+import feign.slf4j.Slf4jLogger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+
+@Service
+public class YoutubeSubscriptionService {
+
+    private static final String CHANNEL_URL = "https://www.youtube.com/xml/feeds/videos.xml?channel_id=";
+
+    private YoutubeSubscriptionRepository repository;
+    private final YoutubeSubscriptionApi youtubeSubscriptionApi;
+    private final BumblebeeConfig config;
+    private List<Subscription> subscriptionList;
+
+    @Autowired
+    public YoutubeSubscriptionService(YoutubeSubscriptionRepository repository,
+                                      BumblebeeConfig config) {
+        this.config = config;
+        this.repository = repository;
+        this.youtubeSubscriptionApi = Feign.builder()
+                .logLevel(feign.Logger.Level.BASIC)
+                .logger(new Slf4jLogger())
+                .target(YoutubeSubscriptionApi.class, YoutubeSubscriptionApi.API_ROOT);
+        this.subscriptionList = retrieveSubscriptions();
+    }
+
+    public void storeSubscription(Subscription subscription) {
+        repository.save(subscription);
+    }
+
+    public List<Subscription> retrieveSubscriptions() {
+        return Lists.newArrayList(repository.findAll());
+    }
+
+    public void deleteSubscription(Subscription subscription) {
+        repository.delete(subscription);
+    }
+
+    public Boolean subscribeChannel(String channelId) {
+        Response response = youtubeSubscriptionApi.subscribe("subscribe", CHANNEL_URL + channelId, config.getUrl() + "/youtube");
```

Comment:
Subscription endpoint could be cached (at least just not duplicate string literals).